### PR TITLE
Simple rails requirement that Dependabot can update

### PR DESCRIPTION
**Pull request by etiennebarrie**
Replaces #495

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,7 +9,12 @@ gem "mocha"
 gem "pg"
 gem "pry-byebug"
 gem "puma"
-gem "rails", defined?(@rails_gem_requirement) && @rails_gem_requirement
+if defined?(@rails_gem_requirement) && @rails_gem_requirement
+  rails = "rails" # confuses Dependabot
```
Comment (adrianna-chang-shopify) [inline Gemfile:13]:
```suggestion
  rails = "rails" # allows Dependabot to update rails properly
```
Nit: I interpret "confuses" as this causing issues for Dependabot, when really we're "tricking" Dependabot into doing the right thing

**Review by adrianna-chang-shopify — COMMENTED** [review]
Can you explain to me a bit more about how this works? Does parsing `rails = "rails"` result in the AST parser skipping the next line? 

Were you able to test this locally?

**Comment by etiennebarrie** [comment]
> Can you explain to me a bit more about how this works? Does parsing `rails = "rails"` result in the AST parser skipping the next line?

Yes, so usually Dependabot doesn't bother updating the Gemfile if it can avoid to (like there's no requirement) but it seems like our code triggers that anyway, based on the backtrace I shared in #495. In any case, once you're in the RequirementReplacer class, it will [look for a call to "gem"](https://github.com/dependabot/dependabot-core/blob/main/bundler/lib/dependabot/bundler/file_updater/requirement_replacer.rb#L124) whose [first parameter is a string of the name of the dependency](https://github.com/dependabot/dependabot-core/blob/18785ae20d06c4c86e794a8c6542aa0c110afc20/bundler/lib/dependabot/bundler/file_updater/requirement_replacer.rb#L133) it's trying to update. Using a local variable is enough to have it not consider this "gem" line at all.

To run it I basically tweaked RequirementReplacer until I could run it on my file with the rails dependency and version 6.1.4.1. I had the same error and then I made these changes and it updated fine (adding the requirement to the Gemfile as a side-effect, I think it's fine if it has to since it's able to update it later, but I'm not even sure it will do it once shipped.

File: Gemfile (hunk 1 above)
Comment (etiennebarrie) [reply Gemfile:13]: Yeah we're tricking it into doing the right thing by confusing it on this line.

**Pull request merged**